app/src/main/java/intellistart/interviewplanning/model/slot/SlotService.kt

```diff
@@ -4,22 +4,24 @@ import intellistart.interviewplanning.exceptions.SlotException
 import intellistart.interviewplanning.exceptions.SlotException.SlotExceptionProfile
 import org.bson.types.ObjectId
 import org.springframework.stereotype.Service
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
 import java.time.LocalDate
 
 @Service
 class SlotService(
     private val slotRepository: SlotRepository
 ) {
 
-    fun create(slot: Slot, email: String): Slot = slotRepository.save(slot, email)
+    fun create(slot: Slot, email: String): Mono<Slot> = slotRepository.save(slot, email)
 
-    fun update(slot: Slot, email: String): Slot = slotRepository.update(slot, email)
+    fun update(slot: Slot, email: String): Mono<Slot> = slotRepository.update(slot, email)
 
-    fun getSlotsByEmailAndDate(email: String, date: LocalDate): List<Slot> =
+    fun getSlotsByEmailAndDate(email: String, date: LocalDate): Flux<Slot> =
         slotRepository.findByEmailAndDate(email, date)
 
-    fun getAllSlotsByEmail(email: String): List<Slot> = slotRepository.findByEmail(email)
+    fun getAllSlotsByEmail(email: String): Flux<Slot> = slotRepository.findByEmail(email)
 
-    fun getById(id: ObjectId): Slot = slotRepository.findById(id)
-        ?: (throw SlotException(SlotExceptionProfile.SLOT_NOT_FOUND))
+    fun getById(id: ObjectId): Mono<Slot> = slotRepository.findById(id)
+        .switchIfEmpty(Mono.error(SlotException(SlotExceptionProfile.SLOT_NOT_FOUND)))
 }
```

Comment:
consider making it deferred:
`.switchIfEmpty{Mono.error(SlotException(SlotExceptionProfile.SLOT_NOT_FOUND))}`
make sure changing it everywhere where you are throwing custom errors